src/lib/components/Flowbite/ThemeContext.tsx

```diff
@@ -30,21 +30,17 @@ export function useTheme(): ThemeContextProps {
   return useContext(ThemeContext);
 }
 
-export const useThemeMode = (
-  usePreferences: boolean,
-): [Mode, React.Dispatch<React.SetStateAction<Mode>>, () => void] => {
-  const [mode, setModeState] = useState<Mode>('light');
-  const savePreference = (mode: Mode) => localStorage.setItem('theme', mode);
-  const getPreference = (): Mode => (localStorage.getItem('theme') as Mode) || getPrefersColorScheme();
+export const useThemeMode = (): [Mode, React.Dispatch<React.SetStateAction<Mode>>, () => void] => {
   const userPreferenceIsDark = () => window.matchMedia?.('(prefers-color-scheme: dark)').matches;
   const getPrefersColorScheme = (): Mode => (userPreferenceIsDark() ? 'dark' : 'light');
-  const toggleMode = () => {
+  const _toggleMode = () => {
```

Comment:
Can you explain why these `_toggleMode` and `_mode` exist in addition to `toggleMode` and `mode`?

Reply:
I can rename it if it's confusing. There's already a reference to component level state `mode` on Line 42. Basically it needs a way to sync with the value of context level `mode`. `_toggleMode` is there to pass the toggle mode functionality to context state

Reply:
That makes sense, but if we could find a way to make this more readable I think it would be immensely helpful for others in the future

Reply:
Let me see if I can fix it quick